### ISSUE TITLE
cookiecutter-2.1.1 breaks cruft

### ISSUE THREAD

**Issue by jlec** [cruft/cruft#166]
After recent upgrade I get

```
➜ cruft update --allow-untracked-files --skip-apply-ask --cookiecutter-input
Traceback (most recent call last):
  File "/Users/justin/.pyenv/versions/dev/bin/cruft", line 8, in <module>
    sys.exit(app())
  File "/Users/justin/.pyenv/versions/dev/lib/python3.10/site-packages/typer/main.py", line 214, in __call__
    return get_command(self)(*args, **kwargs)
  File "/Users/justin/.pyenv/versions/dev/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/Users/justin/.pyenv/versions/dev/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/Users/justin/.pyenv/versions/dev/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/justin/.pyenv/versions/dev/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/justin/.pyenv/versions/dev/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Users/justin/.pyenv/versions/dev/lib/python3.10/site-packages/typer/main.py", line 500, in wrapper
    return callback(**use_params)  # type: ignore
  File "/Users/justin/.pyenv/versions/dev/lib/python3.10/site-packages/cruft/_cli.py", line 237, in update
    if not _commands.update(
  File "/Users/justin/.pyenv/versions/dev/lib/python3.10/site-packages/cruft/_commands/utils/__init__.py", line 14, in wrapper
    return f(*args, **kwargs)
  File "/Users/justin/.pyenv/versions/dev/lib/python3.10/site-packages/cruft/_commands/utils/__init__.py", line 14, in wrapper
    return f(*args, **kwargs)
  File "/Users/justin/.pyenv/versions/dev/lib/python3.10/site-packages/cruft/_commands/update.py", line 63, in update
    _ = utils.generate.cookiecutter_template(
  File "/Users/justin/.pyenv/versions/dev/lib/python3.10/site-packages/cruft/_commands/utils/generate.py", line 40, in cookiecutter_template
    context = _generate_output(cruft_state, Path(repo.working_dir), cookiecutter_input, output_dir)
  File "/Users/justin/.pyenv/versions/dev/lib/python3.10/site-packages/cruft/_commands/utils/generate.py", line 65, in _generate_output
    new_context = generate_cookiecutter_context(
  File "/Users/justin/.pyenv/versions/dev/lib/python3.10/site-packages/cruft/_commands/utils/cookiecutter.py", line 90, in generate_cookiecutter_context
    context = generate_context(
  File "/Users/justin/.pyenv/versions/dev/lib/python3.10/site-packages/cookiecutter/generate.py", line 118, in generate_context
    apply_overwrites_to_context(obj, extra_context)
  File "/Users/justin/.pyenv/versions/dev/lib/python3.10/site-packages/cookiecutter/generate.py", line 68, in apply_overwrites_to_context
    raise ValueError(
ValueError: ['.pre-commit-config.yaml', 'cookiecutter.json', 'hooks', 'template', '{{ cookiecutter.github_repo }}'] provided for choice variable _copy_without_render, but the choices are ['.pre-commit-config.yaml', 'cookiecutter.json', 'hooks', 'template', '{{ cookiecutter.github_repo }}'].
```

Downgrading to cookiecutter-1.7.3 fixes the problem.

What additional information do you need?

**Comment by bigbrozer** [comment]
@jlec do you use a public cookiecutter template ? On this case, could you share the URL ? Thx.

**Comment by samj1912** [comment]
Should be fixed with 2.11.0. Please reopen if issue persists.

**Comment by jlec** [comment]
Issue still persists

```
cruft update --allow-untracked-files --skip-apply-ask
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /Users/justin/.pyenv/versions/3.10.5/envs/dev/lib/python3.10/site-packages/cruft/_cli.py:237 in  │
│ update                                                                                           │
│                                                                                                  │
│   234 │   │   ),                                                                                 │
│   235 │   ),                                                                                     │
│   236 ) -> None:                                                                                 │
│ ❱ 237 │   if not _commands.update(                                                               │
│   238 │   │   project_dir=project_dir,                                                           │
│   239 │   │   cookiecutter_input=cookiecutter_input,                                             │
│   240 │   │   skip_apply_ask=skip_apply_ask,                                                     │
│                                                                                                  │
│ ╭──────────────── locals ────────────────╮                                                       │
│ │ allow_untracked_files = True           │                                                       │
│ │              checkout = None           │                                                       │
│ │    cookiecutter_input = False          │                                                       │
│ │           project_dir = PosixPath('.') │                                                       │
│ │        skip_apply_ask = True           │                                                       │
│ │           skip_update = False          │                                                       │
│ │                strict = True           │                                                       │
│ ╰────────────────────────────────────────╯                                                       │
│                                                                                                  │
│ /Users/justin/.pyenv/versions/3.10.5/envs/dev/lib/python3.10/site-packages/cruft/_commands/utils │
│ /__init__.py:14 in wrapper                                                                       │
│                                                                                                  │
│   11 │   │   def decorator(f):                                                                   │
│   12 │   │   │   @wraps(f)                                                                       │
│   13 │   │   │   def wrapper(*args, **kwargs):                                                   │
│ ❱ 14 │   │   │   │   return f(*args, **kwargs)                                                   │
│   15 │   │   │                                                                                   │
│   16 │   │   │   return wrapper                                                                  │
│   17                                                                                             │
│                                                                                                  │
│ ╭────────────────── locals ───────────────────╮                                                  │
│ │   args = ()                                 │                                                  │
│ │      f = <function update at 0x10fec7010>   │                                                  │
│ │ kwargs = {                                  │                                                  │
│ │          │   'project_dir': PosixPath('.'), │                                                  │
│ │          │   'cookiecutter_input': False,   │                                                  │
│ │          │   'skip_apply_ask': True,        │                                                  │
│ │          │   'skip_update': False,          │                                                  │
│ │          │   'checkout': None,              │                                                  │
│ │          │   'strict': True,                │                                                  │
│ │          │   'allow_untracked_files': True  │                                                  │
│ │          }                                  │                                                  │
│ ╰─────────────────────────────────────────────╯                                                  │
│                                                                                                  │
│ /Users/justin/.pyenv/versions/3.10.5/envs/dev/lib/python3.10/site-packages/cruft/_commands/utils │
│ /__init__.py:14 in wrapper                                                                       │
│                                                                                                  │
│   11 │   │   def decorator(f):                                                                   │
│   12 │   │   │   @wraps(f)                                                                       │
│   13 │   │   │   def wrapper(*args, **kwargs):                                                   │
│ ❱ 14 │   │   │   │   return f(*args, **kwargs)                                                   │
│   15 │   │   │                                                                                   │
│   16 │   │   │   return wrapper                                                                  │
│   17                                                                                             │
│                                                                                                  │
│ ╭────────────────── locals ───────────────────╮                                                  │
│ │   args = ()                                 │                                                  │
│ │      f = <function update at 0x10fec6f80>   │                                                  │
│ │ kwargs = {                                  │                                                  │
│ │          │   'project_dir': PosixPath('.'), │                                                  │
│ │          │   'cookiecutter_input': False,   │                                                  │
│ │          │   'skip_apply_ask': True,        │                                                  │
│ │          │   'skip_update': False,          │                                                  │
│ │          │   'checkout': None,              │                                                  │
│ │          │   'strict': True,                │                                                  │
│ │          │   'allow_untracked_files': True  │                                                  │
│ │          }                                  │                                                  │
│ ╰─────────────────────────────────────────────╯                                                  │
│                                                                                                  │
│ /Users/justin/.pyenv/versions/3.10.5/envs/dev/lib/python3.10/site-packages/cruft/_commands/updat │
│ e.py:63 in update                                                                                │
│                                                                                                  │
│    60 │   │   │   # Generate clean outputs via the cookiecutter                                  │
│    61 │   │   │   # from the current cruft state commit of the cookiectter and the updated       │
│    62 │   │   │   # cookiecutter.                                                                │
│ ❱  63 │   │   │   _ = utils.generate.cookiecutter_template(                                      │
│    64 │   │   │   │   output_dir=current_template_dir,                                           │
│    65 │   │   │   │   repo=repo,                                                                 │
│    66 │   │   │   │   cruft_state=cruft_state,                                                   │
│                                                                                                  │
│ ╭─────────────────────────────────────────── locals ───────────────────────────────────────────╮ │
│ │ allow_untracked_files = True                                                                 │ │
│ │              checkout = None                                                                 │ │
│ │    cookiecutter_input = False                                                                │ │
│ │            cruft_file = PosixPath('.cruft.json')                                             │ │
│ │           cruft_state = {                                                                    │ │
│ │                         │   'checkout': None,                                                │ │
│ │                         │   'commit': '1ed671f37c0cbe969553ccf5951caec7155ad9dd',            │ │
│ │                         │   'context': {                                                     │ │
│ │                         │   │   'cookiecutter': {                                            │ │
│ │                         │   │   │   '_copy_without_render': [                                │ │
│ │                         │   │   │   │   '.pre-commit-config.yaml',                           │ │
│ │                         │   │   │   │   'cookiecutter.json',                                 │ │
│ │                         │   │   │   │   'hooks',                                             │ │
│ │                         │   │   │   │   'template',                                          │ │
│ │                         │   │   │   │   'cookiecutter.github_repo'                           │ │
│ │                         │   │   │   ],                                                       │ │
│ │                         │   │   │   '_template':                                             │ │
│ │                         'git@github.com:foo/cookiecutter-cookie.git',                       │ │
│ │                         │   │   │   'cookiecutter_description': 'Plain gitrepo cookie',      │ │
│ │                         │   │   │   'cookiecutter_name': 'Cookiecutter Git Repo',            │ │
│ │                         │   │   │   'cookiecutter_slug': 'cookiecutter_git_repo',            │ │
│ │                         │   │   │   'email': 'foo@bar.com',                               │ │
│ │                         │   │   │   'full_name': 'Foo Bar',                            │ │
│ │                         │   │   │   'github_repo': 'cookiecutter-git-repo'                   │ │
│ │                         │   │   }                                                            │ │
│ │                         │   },                                                               │ │
│ │                         │   'directory': None,                                               │ │
│ │                         │   'template': 'git@github.com:foo/cookiecutter-cookie.git'        │ │
│ │                         }                                                                    │ │
│ │  current_template_dir = PosixPath('/var/folders/4b/mp6t37sn1zj0qs1ndwpssdbr0000gn/T/tmpab5c… │ │
│ │         deleted_paths = set()                                                                │ │
│ │           last_commit = '54cad86fb964766e8346aff44e94ae4e62422aae'                           │ │
│ │      new_template_dir = PosixPath('/var/folders/4b/mp6t37sn1zj0qs1ndwpssdbr0000gn/T/tmpab5c… │ │
│ │           project_dir = PosixPath('.')                                                       │ │
│ │                  repo = <git.repo.base.Repo                                                  │ │
│ │                         '/var/folders/4b/mp6t37sn1zj0qs1ndwpssdbr0000gn/T/tmpab5cdlec/repo/… │ │
│ │              repo_dir = PosixPath('/var/folders/4b/mp6t37sn1zj0qs1ndwpssdbr0000gn/T/tmpab5c… │ │
│ │        skip_apply_ask = True                                                                 │ │
│ │           skip_update = False                                                                │ │
│ │                strict = True                                                                 │ │
│ │                tmpdir = PosixPath('/var/folders/4b/mp6t37sn1zj0qs1ndwpssdbr0000gn/T/tmpab5c… │ │
│ │               tmpdir_ = '/var/folders/4b/mp6t37sn1zj0qs1ndwpssdbr0000gn/T/tmpab5cdlec'       │ │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────╯ │
│                                                                                                  │
│ /Users/justin/.pyenv/versions/3.10.5/envs/dev/lib/python3.10/site-packages/cruft/_commands/utils │
│ /generate.py:40 in cookiecutter_template                                                         │
│                                                                                                  │
│    37 │   repo.head.reset(commit=commit, working_tree=True)                                      │
│    38 │                                                                                          │
│    39 │   assert repo.working_dir is not None  # nosec B101 (allow assert for type checking)     │
│ ❱  40 │   context = _generate_output(cruft_state, Path(repo.working_dir), cookiecutter_input,    │
│    41 │                                                                                          │
│    42 │   # Get all paths that we are supposed to skip before generating the diff and applying   │
│    43 │   skip_paths = _get_skip_paths(cruft_state, pyproject_file)                              │
│                                                                                                  │
│ ╭─────────────────────────────────────────── locals ───────────────────────────────────────────╮ │
│ │             checkout = '1ed671f37c0cbe969553ccf5951caec7155ad9dd'                            │ │
│ │               commit = '1ed671f37c0cbe969553ccf5951caec7155ad9dd'                            │ │
│ │   cookiecutter_input = False                                                                 │ │
│ │          cruft_state = {                                                                     │ │
│ │                        │   'checkout': None,                                                 │ │
│ │                        │   'commit': '1ed671f37c0cbe969553ccf5951caec7155ad9dd',             │ │
│ │                        │   'context': {                                                      │ │
│ │                        │   │   'cookiecutter': {                                             │ │
│ │                        │   │   │   '_copy_without_render': [                                 │ │
│ │                        │   │   │   │   '.pre-commit-config.yaml',                            │ │
│ │                        │   │   │   │   'cookiecutter.json',                                  │ │
│ │                        │   │   │   │   'hooks',                                              │ │
│ │                        │   │   │   │   'template',                                           │ │
│ │                        │   │   │   │   'cookiecutter.github_repo'                            │ │
│ │                        │   │   │   ],                                                        │ │
│ │                        │   │   │   '_template':                                              │ │
│ │                        'git@github.com:foo/cookiecutter-cookie.git',                        │ │
│ │                        │   │   │   'cookiecutter_description': 'Plain gitrepo cookie',       │ │
│ │                        │   │   │   'cookiecutter_name': 'Cookiecutter Git Repo',             │ │
│ │                        │   │   │   'cookiecutter_slug': 'cookiecutter_git_repo',             │ │
│ │                        │   │   │   'email': 'foo@bar.com',                                │ │
│ │                        │   │   │   'full_name': 'Foo Bar',                             │ │
│ │                        │   │   │   'github_repo': 'cookiecutter-git-repo'                    │ │
│ │                        │   │   }                                                             │ │
│ │                        │   },                                                                │ │
│ │                        │   'directory': None,                                                │ │
│ │                        │   'template': 'git@github.com:foo/cookiecutter-cookie.git'         │ │
│ │                        }                                                                     │ │
│ │        deleted_paths = set()                                                                 │ │
│ │           output_dir = PosixPath('/var/folders/4b/mp6t37sn1zj0qs1ndwpssdbr0000gn/T/tmpab5cd… │ │
│ │          project_dir = PosixPath('.')                                                        │ │
│ │       pyproject_file = PosixPath('pyproject.toml')                                           │ │
│ │                 repo = <git.repo.base.Repo                                                   │ │
│ │                        '/var/folders/4b/mp6t37sn1zj0qs1ndwpssdbr0000gn/T/tmpab5cdlec/repo/.… │ │
│ │ update_deleted_paths = True                                                                  │ │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────╯ │
│                                                                                                  │
│ /Users/justin/.pyenv/versions/3.10.5/envs/dev/lib/python3.10/site-packages/cruft/_commands/utils │
│ /generate.py:65 in _generate_output                                                              │
│                                                                                                  │
│    62 ) -> CookiecutterContext:                                                                  │
│    63 │   inner_dir = project_dir / (cruft_state.get("directory") or "")                         │
│    64 │                                                                                          │
│ ❱  65 │   new_context = generate_cookiecutter_context(                                           │
│    66 │   │   cruft_state["template"],                                                           │
│    67 │   │   inner_dir,                                                                         │
│    68 │   │   extra_context=cruft_state["context"]["cookiecutter"],                              │
│                                                                                                  │
│ ╭─────────────────────────────────────────── locals ───────────────────────────────────────────╮ │
│ │ cookiecutter_input = False                                                                   │ │
│ │        cruft_state = {                                                                       │ │
│ │                      │   'checkout': None,                                                   │ │
│ │                      │   'commit': '1ed671f37c0cbe969553ccf5951caec7155ad9dd',               │ │
│ │                      │   'context': {                                                        │ │
│ │                      │   │   'cookiecutter': {                                               │ │
│ │                      │   │   │   '_copy_without_render': [                                   │ │
│ │                      │   │   │   │   '.pre-commit-config.yaml',                              │ │
│ │                      │   │   │   │   'cookiecutter.json',                                    │ │
│ │                      │   │   │   │   'hooks',                                                │ │
│ │                      │   │   │   │   'template',                                             │ │
│ │                      │   │   │   │   'cookiecutter.github_repo'                              │ │
│ │                      │   │   │   ],                                                          │ │
│ │                      │   │   │   '_template': 'git@github.com:foo/cookiecutter-cookie.git', │ │
│ │                      │   │   │   'cookiecutter_description': 'Plain gitrepo cookie',         │ │
│ │                      │   │   │   'cookiecutter_name': 'Cookiecutter Git Repo',               │ │
│ │                      │   │   │   'cookiecutter_slug': 'cookiecutter_git_repo',               │ │
│ │                      │   │   │   'email': 'foo@bar.com',                                  │ │
│ │                      │   │   │   'full_name': 'Foo Bar',                               │ │
│ │                      │   │   │   'github_repo': 'cookiecutter-git-repo'                      │ │
│ │                      │   │   }                                                               │ │
│ │                      │   },                                                                  │ │
│ │                      │   'directory': None,                                                  │ │
│ │                      │   'template': 'git@github.com:foo/cookiecutter-cookie.git'           │ │
│ │                      }                                                                       │ │
│ │          inner_dir = PosixPath('/var/folders/4b/mp6t37sn1zj0qs1ndwpssdbr0000gn/T/tmpab5cdle… │ │
│ │         output_dir = PosixPath('/var/folders/4b/mp6t37sn1zj0qs1ndwpssdbr0000gn/T/tmpab5cdle… │ │
│ │        project_dir = PosixPath('/var/folders/4b/mp6t37sn1zj0qs1ndwpssdbr0000gn/T/tmpab5cdle… │ │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────╯ │
│                                                                                                  │
│ /Users/justin/.pyenv/versions/3.10.5/envs/dev/lib/python3.10/site-packages/cruft/_commands/utils │
│ /cookiecutter.py:90 in generate_cookiecutter_context                                             │
│                                                                                                  │
│    87 │   │   config_file=str(config_file) if config_file else None, default_config=default_co   │
│    88 │   )                                                                                      │
│    89 │                                                                                          │
│ ❱  90 │   context = generate_context(                                                            │
│    91 │   │   context_file=context_file,                                                         │
│    92 │   │   default_context=config_dict["default_context"],                                    │
│    93 │   │   extra_context=extra_context,                                                       │
│                                                                                                  │
│ ╭─────────────────────────────────────────── locals ───────────────────────────────────────────╮ │
│ │               config_dict = {                                                                │ │
│ │                             │   'cookiecutters_dir': '/Users/justin/.cookiecutters/',        │ │
│ │                             │   'replay_dir': '/Users/justin/.cookiecutter_replay/',         │ │
│ │                             │   'default_context': OrderedDict([('full_name', 'Foo        │ │
│ │                             Foo'), ('email', 'foo@bar.com'), ('github_username',       │ │
│ │                             'jlec'), ('license', 'Apache Software License 2.0'),             │ │
│ │                             ('ansible_namespace', 'jlec')]),                                 │ │
│ │                             │   'abbreviations': {                                           │ │
│ │                             │   │   'gh': 'https://github.com/{0}.git',                      │ │
│ │                             │   │   'gl': 'https://gitlab.com/{0}.git',                      │ │
│ │                             │   │   'bb': 'https://bitbucket.org/{0}'                        │ │
│ │                             │   }                                                            │ │
│ │                             }                                                                │ │
│ │               config_file = None                                                             │ │
│ │              context_file = PosixPath('/var/folders/4b/mp6t37sn1zj0qs1ndwpssdbr0000gn/T/tmp… │ │
│ │ cookiecutter_template_dir = PosixPath('/var/folders/4b/mp6t37sn1zj0qs1ndwpssdbr0000gn/T/tmp… │ │
│ │            default_config = False                                                            │ │
│ │             extra_context = {                                                                │ │
│ │                             │   '_copy_without_render': [                                    │ │
│ │                             │   │   '.pre-commit-config.yaml',                               │ │
│ │                             │   │   'cookiecutter.json',                                     │ │
│ │                             │   │   'hooks',                                                 │ │
│ │                             │   │   'template',                                              │ │
│ │                             │   │   'cookiecutter.github_repo'                               │ │
│ │                             │   ],                                                           │ │
│ │                             │   '_template': 'git@github.com:foo/cookiecutter-cookie.git',  │ │
│ │                             │   'cookiecutter_description': 'Plain gitrepo cookie',          │ │
│ │                             │   'cookiecutter_name': 'Cookiecutter Git Repo',                │ │
│ │                             │   'cookiecutter_slug': 'cookiecutter_git_repo',                │ │
│ │                             │   'email': 'foo@bar.com',                                   │ │
│ │                             │   'full_name': 'Foo Bar',                                │ │
│ │                             │   'github_repo': 'cookiecutter-git-repo'                       │ │
│ │                             }                                                                │ │
│ │                  no_input = True                                                             │ │
│ │          template_git_url = 'git@github.com:foo/cookiecutter-cookie.git'                    │ │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────╯ │
│                                                                                                  │
│ /Users/justin/.pyenv/versions/3.10.5/envs/dev/lib/python3.10/site-packages/cookiecutter/generate │
│ .py:117 in generate_context                                                                      │
│                                                                                                  │
│   114 │   │   except ValueError as ex:                                                           │
│   115 │   │   │   warnings.warn("Invalid default received: " + str(ex))                          │
│   116 │   if extra_context:                                                                      │
│ ❱ 117 │   │   apply_overwrites_to_context(obj, extra_context)                                    │
│   118 │                                                                                          │
│   119 │   logger.debug('Context generated is %s', context)                                       │
│   120 │   return context                                                                         │
│                                                                                                  │
│ ╭─────────────────────────────────────────── locals ───────────────────────────────────────────╮ │
│ │         context = OrderedDict([('cookiecutter', OrderedDict([('_copy_without_render',        │ │
│ │                   ['.pre-commit-config.yaml', 'cookiecutter.json', 'hooks', 'template',      │ │
│ │                   'cookiecutter.github_repo']), ('cookiecutter_description', 'Awesome        │ │
│ │                   cookiecutter inception'), ('cookiecutter_name', 'Cookiecutter Cookies'),   │ │
│ │                   ('cookiecutter_slug', "{{ cookiecutter.cookiecutter_name.lower().replace(' │ │
│ │                   ', '_').replace('-', '_') }}"), ('email', 'foo@bar.com'), ('full_name', │ │
│ │                   'Foo Bar'), ('github_repo', '{{ cookiecutter.cookiecutter_slug       │ │
│ │                   }}')]))])                                                                  │ │
│ │    context_file = PosixPath('/var/folders/4b/mp6t37sn1zj0qs1ndwpssdbr0000gn/T/tmpab5cdlec/r… │ │
│ │ default_context = OrderedDict([('full_name', 'Foo Bar'), ('email', 'foo@bar.com'),  │ │
│ │                   ('github_username', 'jlec'), ('license', 'Apache Software License 2.0'),   │ │
│ │                   ('ansible_namespace', 'jlec')])                                            │ │
│ │   extra_context = {                                                                          │ │
│ │                   │   '_copy_without_render': [                                              │ │
│ │                   │   │   '.pre-commit-config.yaml',                                         │ │
│ │                   │   │   'cookiecutter.json',                                               │ │
│ │                   │   │   'hooks',                                                           │ │
│ │                   │   │   'template',                                                        │ │
│ │                   │   │   'cookiecutter.github_repo'                                         │ │
│ │                   │   ],                                                                     │ │
│ │                   │   '_template': 'git@github.com:foo/cookiecutter-cookie.git',            │ │
│ │                   │   'cookiecutter_description': 'Plain gitrepo cookie',                    │ │
│ │                   │   'cookiecutter_name': 'Cookiecutter Git Repo',                          │ │
│ │                   │   'cookiecutter_slug': 'cookiecutter_git_repo',                          │ │
│ │                   │   'email': 'foo@bar.com',                                             │ │
│ │                   │   'full_name': 'Foo Bar',                                          │ │
│ │                   │   'github_repo': 'cookiecutter-git-repo'                                 │ │
│ │                   }                                                                          │ │
│ │     file_handle = <_io.TextIOWrapper                                                         │ │
│ │                   name='/var/folders/4b/mp6t37sn1zj0qs1ndwpssdbr0000gn/T/tmpab5cdlec/repo/c… │ │
│ │                   mode='r' encoding='utf-8'>                                                 │ │
│ │       file_name = 'cookiecutter.json'                                                        │ │
│ │       file_stem = 'cookiecutter'                                                             │ │
│ │             obj = OrderedDict([('_copy_without_render', ['.pre-commit-config.yaml',          │ │
│ │                   'cookiecutter.json', 'hooks', 'template', 'cookiecutter.github_repo']),    │ │
│ │                   ('cookiecutter_description', 'Awesome cookiecutter inception'),            │ │
│ │                   ('cookiecutter_name', 'Cookiecutter Cookies'), ('cookiecutter_slug', "{{   │ │
│ │                   cookiecutter.cookiecutter_name.lower().replace(' ', '_').replace('-', '_') │ │
│ │                   }}"), ('email', 'foo@bar.com'), ('full_name', 'Foo Bar'),         │ │
│ │                   ('github_repo', '{{ cookiecutter.cookiecutter_slug }}')])                  │ │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────╯ │
│                                                                                                  │
│ /Users/justin/.pyenv/versions/3.10.5/envs/dev/lib/python3.10/site-packages/cookiecutter/generate │
│ .py:67 in apply_overwrites_to_context                                                            │
│                                                                                                  │
│    64 │   │   │   │   context_value.remove(overwrite)                                            │
│    65 │   │   │   │   context_value.insert(0, overwrite)                                         │
│    66 │   │   │   else:                                                                          │
│ ❱  67 │   │   │   │   raise ValueError(                                                          │
│    68 │   │   │   │   │   "{} provided for choice variable {}, but the "                         │
│    69 │   │   │   │   │   "choices are {}.".format(overwrite, variable, context_value)           │
│    70 │   │   │   │   )                                                                          │
│                                                                                                  │
│ ╭─────────────────────────────────────────── locals ───────────────────────────────────────────╮ │
│ │           context = OrderedDict([('_copy_without_render', ['.pre-commit-config.yaml',        │ │
│ │                     'cookiecutter.json', 'hooks', 'template', 'cookiecutter.github_repo']),  │ │
│ │                     ('cookiecutter_description', 'Awesome cookiecutter inception'),          │ │
│ │                     ('cookiecutter_name', 'Cookiecutter Cookies'), ('cookiecutter_slug', "{{ │ │
│ │                     cookiecutter.cookiecutter_name.lower().replace(' ', '_').replace('-',    │ │
│ │                     '_') }}"), ('email', 'foo@bar.com'), ('full_name', 'Foo Bar'),  │ │
│ │                     ('github_repo', '{{ cookiecutter.cookiecutter_slug }}')])                │ │
│ │     context_value = [                                                                        │ │
│ │                     │   '.pre-commit-config.yaml',                                           │ │
│ │                     │   'cookiecutter.json',                                                 │ │
│ │                     │   'hooks',                                                             │ │
│ │                     │   'template',                                                          │ │
│ │                     │   'cookiecutter.github_repo'                                           │ │
│ │                     ]                                                                        │ │
│ │         overwrite = [                                                                        │ │
│ │                     │   '.pre-commit-config.yaml',                                           │ │
│ │                     │   'cookiecutter.json',                                                 │ │
│ │                     │   'hooks',                                                             │ │
│ │                     │   'template',                                                          │ │
│ │                     │   'cookiecutter.github_repo'                                           │ │
│ │                     ]                                                                        │ │
│ │ overwrite_context = {                                                                        │ │
│ │                     │   '_copy_without_render': [                                            │ │
│ │                     │   │   '.pre-commit-config.yaml',                                       │ │
│ │                     │   │   'cookiecutter.json',                                             │ │
│ │                     │   │   'hooks',                                                         │ │
│ │                     │   │   'template',                                                      │ │
│ │                     │   │   'cookiecutter.github_repo'                                       │ │
│ │                     │   ],                                                                   │ │
│ │                     │   '_template': 'git@github.com:foo/cookiecutter-cookie.git',          │ │
│ │                     │   'cookiecutter_description': 'Plain gitrepo cookie',                  │ │
│ │                     │   'cookiecutter_name': 'Cookiecutter Git Repo',                        │ │
│ │                     │   'cookiecutter_slug': 'cookiecutter_git_repo',                        │ │
│ │                     │   'email': 'foo@bar.com',                                           │ │
│ │                     │   'full_name': 'Foo Bar',                                        │ │
│ │                     │   'github_repo': 'cookiecutter-git-repo'                               │ │
│ │                     }                                                                        │ │
│ │          variable = '_copy_without_render'                                                   │ │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────╯ │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
ValueError: ['.pre-commit-config.yaml', 'cookiecutter.json', 'hooks', 'template', 'cookiecutter.github_repo'] provided for choice variable
_copy_without_render, but the choices are ['.pre-commit-config.yaml', 'cookiecutter.json', 'hooks', 'template', 'cookiecutter.github_repo'].
```

**Comment by jlec** [comment]
Take [jlec/cookiecutter-test](https://github.com/jlec/cookiecutter-test.git). 

Make sure you create the repo from the base commit and then update to the latest. The bug is only trigger in the update call.

**Comment by simonwardjones** [comment]
I am having this problem too. I think the `apply_overwrites_to_context` function within cookiecutter is assuming that the overwrite variable is never also a list. I think this will need to be fixed on the cookiecutter side so that in the case when the overwrite is also a list it just overwrites.

**Comment by adamcunnington-mlg** [comment]
likewise. I guess we are waiting on the cookiecutter issue to be closed still - I chased it

**Comment by luisdavim** [comment]
same issue here:

```
Traceback (most recent call last):

  File "/usr/local/bin/cruft", line 8, in <module>
    sys.exit(app())

  File "/usr/local/lib/python3.10/site-packages/cruft/_cli.py", line 237, in update
    if not _commands.update(

  File "/usr/local/lib/python3.10/site-packages/cruft/_commands/utils/__init__.py", line 14, in wrapper
    return f(*args, **kwargs)

  File "/usr/local/lib/python3.10/site-packages/cruft/_commands/utils/__init__.py", line 14, in wrapper
    return f(*args, **kwargs)

  File "/usr/local/lib/python3.10/site-packages/cruft/_commands/update.py", line 63, in update
    _ = utils.generate.cookiecutter_template(

  File "/usr/local/lib/python3.10/site-packages/cruft/_commands/utils/generate.py", line 40, in cookiecutter_template
    context = _generate_output(cruft_state, Path(repo.working_dir), cookiecutter_input, output_dir)

  File "/usr/local/lib/python3.10/site-packages/cruft/_commands/utils/generate.py", line 65, in _generate_output
    new_context = generate_cookiecutter_context(

  File "/usr/local/lib/python3.10/site-packages/cruft/_commands/utils/cookiecutter.py", line 90, in generate_cookiecutter_context
    context = generate_context(

  File "/usr/local/lib/python3.10/site-packages/cookiecutter/generate.py", line 117, in generate_context
    apply_overwrites_to_context(obj, extra_context)

  File "/usr/local/lib/python3.10/site-packages/cookiecutter/generate.py", line 67, in apply_overwrites_to_context
    raise ValueError(

ValueError: ['.chglog/CHANGELOG.tpl.md', '.github/workflows/terraform.yml'] provided for choice variable _copy_without_render, but the choices are ['.chglog/CHANGELOG.tpl.md', '.github/workflows/terraform.yml'].
```

**Comment by OverkillGuy** [comment]
Same error on my side but due to the `_extensions` key (I use the `jinja2-git` extension to include `{% gitcommit %}` in the template, loving it).

```
Traceback (most recent call last):

  File "/home/jiby/.local/bin/cruft", line 8, in <module>
    sys.exit(app())

  File "/home/jiby/.local/pipx/venvs/cruft/lib/python3.10/site-packages/cruft/_cli.py", line 237, in update
    if not _commands.update(

  File "/home/jiby/.local/pipx/venvs/cruft/lib/python3.10/site-packages/cruft/_commands/utils/__init__.py", line 14, in wrapper
    return f(*args, **kwargs)

  File "/home/jiby/.local/pipx/venvs/cruft/lib/python3.10/site-packages/cruft/_commands/utils/__init__.py", line 14, in wrapper
    return f(*args, **kwargs)

  File "/home/jiby/.local/pipx/venvs/cruft/lib/python3.10/site-packages/cruft/_commands/update.py", line 63, in update
    _ = utils.generate.cookiecutter_template(

  File "/home/jiby/.local/pipx/venvs/cruft/lib/python3.10/site-packages/cruft/_commands/utils/generate.py", line 40, in cookiecutter_template
    context = _generate_output(cruft_state, Path(repo.working_dir), cookiecutter_input, output_dir)

  File "/home/jiby/.local/pipx/venvs/cruft/lib/python3.10/site-packages/cruft/_commands/utils/generate.py", line 65, in _generate_output
    new_context = generate_cookiecutter_context(

  File "/home/jiby/.local/pipx/venvs/cruft/lib/python3.10/site-packages/cruft/_commands/utils/cookiecutter.py", line 90, in generate_cookiecutter_context
    context = generate_context(

  File "/home/jiby/.local/pipx/venvs/cruft/lib/python3.10/site-packages/cookiecutter/generate.py", line 117, in generate_context
    apply_overwrites_to_context(obj, extra_context)

  File "/home/jiby/.local/pipx/venvs/cruft/lib/python3.10/site-packages/cookiecutter/generate.py", line 67, in apply_overwrites_to_context
    raise ValueError(

ValueError: ['jinja2_git.GitExtension'] provided for choice variable _extensions, but the choices are ['jinja2_git.GitExtension'].
```

That's during a `cruft update` while adopting cruft itself on an existing cookiecutter template.

**Comment by OverkillGuy** [comment]
Hello, as mentioned I created #197 to address the issue of ValueError on cookiecutter.json keys with underscore prefix.

Keen to hear how all the participants of this thread experience my PR, at least until upstream fixes/accept fix linked.

**Comment by lyz-code** [comment]
We're also seeing this issue, what can we do to close this issue?

**Comment by jslorrma** [comment]
Hello cruft-team ...

I really like your project. I was halfway in implementing something like `cruft` on my own when I found this. It's really great work and saves me a lot of time. Thanks for that ... 👍👍

In one of my `cookiecutter` templates I'm using `'_extensions'` and see a similar error as reported here.

**cookiecutter.json**:
{
    "project_name": "The name of the python library",
    ...
    "_extensions": ["jinja2_time.TimeExtension"],
}

**error**:
```python 
ValueError: ['jinja2_time.TimeExtension'] provided for choice variable _extensions, but the choices are ['jinja2_time.TimeExtension'].
```

Any update on #197? Tested the [fix provided in #197](https://github.com/cruft/cruft/pull/197/files) locally. It solves my issue.
